### ISSUE TITLE
Scripting does not really works

### ISSUE THREAD

**Issue by lanxx019** [database-rider/database-rider#46]
I have the following yaml file:

```
shipments:
  - id: 1
    uuid: kotlin:java.util.UUID.randomUUID()
    shipper_application_id: 1
    carrier_application_id: 1
    driver_id: 2
    state: "in_transit"
    updated_at: kotlin:java.sql.Timestamp.from(java.time.Instant.now())
    created_by_id: 1
  - id: 2
    uuid: kotlin:java.util.UUID.randomUUID()
    shipper_application_id: 1
    carrier_application_id: 1
    driver_id: 3
    state: "assigned"
    updated_at: kotlin:java.sql.Timestamp.from(java.time.Instant.now())
    created_by_id: 1
```

with this given, I would expect that the `kotlin:java.util.UUID.randomUUID()` call is evaluated everytime, but I am getting this:

```
Caused by: org.dbunit.DatabaseUnitException: Exception processing table name='shipments'
	at org.dbunit.operation.AbstractBatchOperation.execute(AbstractBatchOperation.java:240)
	at org.dbunit.operation.CompositeOperation.execute(CompositeOperation.java:79)
	at com.github.database.rider.core.dataset.DataSetExecutorImpl.createDataSet(DataSetExecutorImpl.java:170)
	... 40 more
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "index_shipments_on_uuid"
  Detail: Key (uuid)=(e2d001da-722d-41ec-b41b-479c1e392abe) already exists.
	at org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2477)
	at org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2190)
	at org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:300)
	at org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:428)
	at org.postgresql.jdbc.PgStatement.execute(PgStatement.java:354)
	at org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:169)
	at org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:158)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at org.dbunit.database.statement.SimplePreparedStatement.addBatch(SimplePreparedStatement.java:80)
	at org.dbunit.database.statement.AutomaticPreparedBatchStatement.addBatch(AutomaticPreparedBatchStatement.java:70)
	at org.dbunit.operation.AbstractBatchOperation.execute(AbstractBatchOperation.java:224)
	... 42 more
```

I also verified this with the built-in `js` engine as well. Is this expected?

**Comment by rmpestano** [comment]
Hi @lanxx019,

can you try with:

```
uuid: groovy: new Random()
``` 

You'll need groovy on your test classpath:

```
<dependency>
    <groupId>org.codehaus.groovy</groupId>
    <artifactId>groovy-all</artifactId>
    <version>2.4.6</version>
    <scope>test</scope>
</dependency>
```

or with js:

```
uuid: js: Math.random()
``` 
kotlin is not support as a script engine.

**Comment by rmpestano** [comment]
Also there is some docs which may help here: https://database-rider.github.io/database-rider/1.2.0/documentation.html#Dynamic-data-using-scritable-datasets

**Comment by lanxx019** [comment]
> or with js:

`js:Mathh.random()` gives the same result:

```
Caused by: org.postgresql.util.PSQLException: ERROR: duplicate key value violates unique constraint "index_shipments_on_uuid"
  Detail: Key (uuid)=(0.8652061650663581) already exists.
```
All the timestamp I created with different rows also show the exact same value.

> kotlin is not support as a script engine.

Does `database-rider` only support two implementations of JSR-223 or any scripting language supports JSR-223 should work? This is confusing since my other settings are working fine with `kotlin`.

> Also there is some docs which may help here

 I do not think the test cases covers the problem I encountered since The test only insert one row into DB.

**Comment by rmpestano** [comment]
Hi, you are right about kotlin, didnt knew It had an impl for jsr 223.

About the random issue I'll investigate.

**Comment by rmpestano** [comment]
Hi @lanxx019,

can you try with latest snapshot?

```
 <dependency>
            <groupId>com.github.database-rider</groupId>
            <artifactId>rider-core</artifactId>
            <version>1.2.3-SNAPSHOT</version>
            <scope>test</scope>
        </dependency>
```

You'll have to add snapshots repository on your pom.xml:

```
<repositories>
    <repository>
        <snapshots/>
        <id>snapshots</id>
        <name>libs-snapshot</name>
        <url>https://oss.sonatype.org/content/repositories/snapshots</url>
    </repository>
</repositories>
```

**Comment by lanxx019** [comment]
OK, this works now. Thanks.

**Comment by rmpestano** [comment]
That's good knews. I've just released v1.2.3 with this fix.

**Comment by renannprado** [comment]
It doesn't work for me. I have the below definition:

```yaml
table:
  - id: groovy:java.util.UUID.randomUUID().toString()
```

I've added the below dependency:

```gradle
testImplementation("org.codehaus.groovy:groovy-all:2.4.6")
```

And got this error:

```
Caused by: javax.script.ScriptException: groovy.lang.MissingMethodException: No signature of method: static java.util.UUID.randomUUID() is applicable for argument types: () values: []
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:355)
	at org.codehaus.groovy.jsr223.GroovyScriptEngineImpl.eval(GroovyScriptEngineImpl.java:153)
```

I tried different combinations like:

* `id: "groovy: java.util.UUID.randomUUID().toString()"`
* `id: groovy:java.util.UUID.randomUUID().toString()`
* `id: groovy:java.util.UUID.randomUUID()`
* `id: groovy:UUID.randomUUID()`
* `id: groovy:UUID.randomUUID().toString()`

And none of them worked.

I've also tried to use `kotlin` (which I would prefer) by adding the below dependency:

```kotlin
table:
  - id: kotlin:java.util.UUID.randomUUID().toString()
```

Added the below dependency:

```gradle
testImplementation("org.jetbrains.kotlin:kotlin-scripting-jsr223")
```

and still got this error:

```
Caused by: org.postgresql.util.PSQLException: ERROR: invalid input syntax for type uuid: "kotlin:java.util.UUID.randomUUID().toString()"
	at app//org.postgresql.core.v3.QueryExecutorImpl.receiveErrorResponse(QueryExecutorImpl.java:2676)
	at app//org.postgresql.core.v3.QueryExecutorImpl.processResults(QueryExecutorImpl.java:2366)
	at app//org.postgresql.core.v3.QueryExecutorImpl.execute(QueryExecutorImpl.java:356)
	at app//org.postgresql.jdbc.PgStatement.executeInternal(PgStatement.java:496)
	at app//org.postgresql.jdbc.PgStatement.execute(PgStatement.java:413)
	at app//org.postgresql.jdbc.PgPreparedStatement.executeWithFlags(PgPreparedStatement.java:190)
	at app//org.postgresql.jdbc.PgPreparedStatement.execute(PgPreparedStatement.java:177)
	at app//com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at app//com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at app//org.dbunit.database.statement.SimplePreparedStatement.addBatch(SimplePreparedStatement.java:85)
	at app//org.dbunit.database.statement.AutomaticPreparedBatchStatement.addBatch(AutomaticPreparedBatchStatement.java:70)
	at app//org.dbunit.operation.AbstractBatchOperation.execute(AbstractBatchOperation.java:224)
	... 72 more
```

And then I gave up, until someone can shad some light here for me.

@lanxx019 could you maybe post did you manage to make it work? or @rmpestano do you know? I looked for documentation and could not find any. If kotlin works it would be even better, but I would happy with groovy already.

I'm using the latest version `1.38.1` of database-rider.
